src/main/java/com/dduru/gildongmu/verification/service/PhoneVerificationService.java

```diff
@@ -0,0 +1,68 @@
+package com.dduru.gildongmu.verification.service;
+
+import com.dduru.gildongmu.common.jwt.JwtTokenProvider;
+import com.dduru.gildongmu.user.repository.UserRepository;
+import com.dduru.gildongmu.verification.dto.VerificationSendResponse;
+import com.dduru.gildongmu.verification.dto.VerificationVerifyResponse;
+import com.dduru.gildongmu.verification.exception.DuplicatePhoneNumberException;
+import com.dduru.gildongmu.verification.exception.SmsProviderException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class PhoneVerificationService {
+
+    private final SmsService smsService;
+    private final VerificationCodeService verificationCodeService;
+    private final JwtTokenProvider jwtTokenProvider;
+    private final UserRepository userRepository;
+
+    public VerificationSendResponse sendVerificationCode(String phoneNumber) {
+        validatePhoneNumber(phoneNumber);
+
+        VerificationCodeService.VerificationCreateResult result = 
+                verificationCodeService.createVerification(phoneNumber);
+
+        try {
+            sendVerificationSms(phoneNumber, result.code());
+            verificationCodeService.setResendLimit(phoneNumber);
+        } catch (Exception e) {
+            verificationCodeService.rollbackVerificationCreation(phoneNumber);
+            throw e;
```

Comment:
에러 발생 시 디버깅이 수월하도록 아래처럼 예외처리를 분명하게 구분하면 더 좋을 거 같습니다.

의미 있는 도메인 예외로 감싸서 던지면
로그와 API 레벨에서 책임이 분리되어 보일 것 같아요.

```suggestion
        } catch (Exception e) {
            verificationCodeService.rollbackVerificationCreation(phoneNumber);
            log.error("Verification code 메서드 처리 실패", e);
            throw new SmsServiceFailureException("SMS 발송 실패 등 서비스 처리 중 문제 발생", e);
        }
```